x-pack/plugins/watcher/__jest__/client_integration/watch_create_threshold.test.tsx

```diff
@@ -753,20 +765,74 @@ describe('<ThresholdWatchEdit /> create route', () => {
         });
       });
 
+      describe('watch visualize data payload', () => {
+        test('should send the correct payload', async () => {
+          const { form, find, component } = testBed;
+
+          // Set up required fields
+          await act(async () => {
+            form.setInputValue('nameInput', WATCH_NAME);
+            find('indicesComboBox').simulate('change', [{ label: 'index1', value: 'index1' }]);
+            form.setInputValue('watchTimeFieldSelect', WATCH_TIME_FIELD);
+          });
+          component.update();
+
+          const latestReqToGetVisualizeData = server.requests.find(
+            (req) => req.method === 'POST' && req.url === '/api/watcher/watch/visualize'
+          );
+          if (!latestReqToGetVisualizeData) {
+            throw new Error(`No request found to fetch visualize data.`);
+          }
+
+          const requestBody = unwrapBodyResponse(latestReqToGetVisualizeData.requestBody);
+
+          expect(requestBody.watch).toEqual({
+            id: requestBody.watch.id, // id is dynamic
+            name: 'my_test_watch',
+            type: 'threshold',
+            isNew: true,
+            isActive: true,
+            actions: [],
+            index: ['index1'],
+            timeField: '@timestamp',
+            triggerIntervalSize: 1,
+            triggerIntervalUnit: 'm',
+            aggType: 'count',
+            termSize: 5,
+            termOrder: 'desc',
+            thresholdComparator: '>',
+            timeWindowSize: 5,
+            timeWindowUnit: 'm',
+            hasTermsAgg: false,
+            threshold: 1000,
+          });
+
+          if (kibanaVersion.major < 8) {
+            // In 7.x we use the deprecated "interval" parameter
+            expect(requestBody.options.interval).toBeDefined();
+            expect(requestBody.options.fixed_interval).not.toBeDefined();
```

Comment:
Nit: I think `toBeUndefined()` is a little more human readable, here and on line 816